src/main/java/com/iexec/worker/amnesia/AmnesiaRecoveryService.java

```diff
@@ -66,151 +66,105 @@ public AmnesiaRecoveryService(CustomFeignClient customFeignClient,
             ContributionAuthorization contributionAuth = interruptedReplicate.getContributionAuthorization();
             RecoveryAction recoveryAction = interruptedReplicate.getRecoveryAction();
             String chainTaskId = contributionAuth.getChainTaskId();
+            boolean isResultAvailable = isResultAvailable(chainTaskId);
 
-            log.info("recovering interrupted task [chainTaskId:{}, recoveryAction:{}]",
+            log.info("Recovering interrupted task [chainTaskId:{}, recoveryAction:{}]",
                     chainTaskId, recoveryAction);
 
-            boolean shouldSubscribe = false;
-            boolean shouldKeepResultFolder = true;
-
-            switch (interruptedReplicate.getRecoveryAction()) {
-                case WAIT:
-                    shouldSubscribe = true; // just subscribe and wait
-                    break;
-
-                case CONTRIBUTE:
-                    shouldSubscribe = recoverReplicateByContributing(contributionAuth);
-                    shouldKeepResultFolder = shouldSubscribe;
-                    break;
-
-                case ABORT_CONSENSUS_REACHED:
-                    taskExecutorService.abortConsensusReached(chainTaskId);
-                    break;
-
-                case ABORT_CONTRIBUTION_TIMEOUT:
-                    taskExecutorService.abortContributionTimeout(chainTaskId);
-                    break;
-
-                case REVEAL:
-                    shouldSubscribe = recoverReplicateByRevealing(contributionAuth);
-                    shouldKeepResultFolder = shouldSubscribe;
-                    break;
-
-                case UPLOAD_RESULT:
-                    shouldSubscribe = recoverReplicateByUploadingResult(contributionAuth);
-                    shouldKeepResultFolder = shouldSubscribe;
-                    break;
-
-                case COMPLETE:
-                    taskExecutorService.completeTask(chainTaskId);
-                    break;
+            if (!isResultAvailable && recoveryAction != RecoveryAction.CONTRIBUTE) {
```

Comment:
What do you think about something like:
```
isResultRequired = (recoveryAction == WAIT ||recoveryAction == CONTRIBUTE || recoveryAction == REVEAL recoveryAction == UPLOAD_RESULT);
if(!isResultAvailable && isResultRequired){
  log.error(..)
  continue;
}
```

Reply:
Yes, it's clearer. But when `recoveryAction == CONTRIBUTE` the result is not necessarily needed because we can re-do the computation.
So I'd say:
`isResultRequired = Arrays.asList(WAIT, REVEAL, UPLOAD_RESULT).contains(recoveryAction);`